src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/GetClaimByReferenceNumberTest.java

```diff
@@ -2,65 +2,41 @@
 import uk.gov.hmcts.cmc.claimstore.models.Claim;
 
-import java.util.Optional;
-
-import static org.mockito.ArgumentMatchers.eq;
-import static org.mockito.BDDMockito.given;
+import static org.assertj.core.api.Assertions.assertThat;
 import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
 
 @ActiveProfiles("test")
-public class GetClaimByReferenceNumberTest extends BaseTest {
-
-    private static final String REFERENCE_NUMBER = "000MC001";
-    private static final String EXTERNAL_ID = "9f49d8df-b734-4e86-aeb6-e22f0c2ca78d";
-
-    private final Claim claim = new Claim.Builder().setId(1L).setSubmitterId(2L).setLetterHolderId(3L)
-        .setExternalId(EXTERNAL_ID)
-        .setReferenceNumber(REFERENCE_NUMBER)
-        .build();
+public class GetClaimByReferenceNumberTest extends BaseIntegrationTest {
 
     @Test
     public void shouldReturn200HttpStatusWhenClaimFound() throws Exception {
+        Claim claim = claimStore.saveClaim(SampleClaimData.builder().build());
 
-        given(claimRepository.getByClaimReferenceNumber(eq(REFERENCE_NUMBER))).willReturn(Optional.of(claim));
-
-        webClient
-            .perform(get("/testing-support/claims/" + REFERENCE_NUMBER))
+        MvcResult result = makeRequest(claim.getReferenceNumber())
             .andExpect(status().isOk())
             .andReturn();
-    }
-
-    @Test
-    public void shouldReturn404HttpStatusWhenNoClaimFound() throws Exception {
 
-        given(claimRepository.getByClaimReferenceNumber(eq(REFERENCE_NUMBER))).willReturn(Optional.empty());
-
-        webClient
-            .perform(get("/testing-support/claims/" + REFERENCE_NUMBER))
-            .andExpect(status().isNotFound())
-            .andReturn();
+        assertThat(deserializeObjectFrom(result, Claim.class))
+            .extracting(Claim::getReferenceNumber).containsExactly(claim.getReferenceNumber());
     }
 
     @Test
-    public void shouldReturn404HttpStatusWhenWrongUrlGiven() throws Exception {
+    public void shouldReturn404HttpStatusWhenNoClaimFound() throws Exception {
+        String nonExistingReferenceNumber = "000MC900";
 
-        webClient
-            .perform(get("/testing-support/not-existing-endpoint"))
-            .andExpect(status().isNotFound())
-            .andReturn();
+        makeRequest(nonExistingReferenceNumber)
+            .andExpect(status().isNotFound());
     }
 
-    @Test
-    public void shouldReturn500HttpStatusWhenInternalErrorOccurs() throws Exception {
-        given(claimRepository.getByClaimReferenceNumber(eq(REFERENCE_NUMBER))).willThrow(new RuntimeException("error"));
-
-        webClient
-            .perform(get("/testing-support/claims/" + REFERENCE_NUMBER))
-            .andExpect(status().isInternalServerError())
-            .andReturn();
+    private ResultActions makeRequest(String referenceNumber) throws Exception {
```

Comment:
why don't we extract this method to some utils?

Reply:
Each test has its own `makeRequest` method which differs slightly by hitting different API paths.